tests/cpptests/test_cpp_query.cpp

```diff
@@ -454,6 +721,61 @@ TEST_F(QueryTest, testFieldSpec) {
   IndexSpec_Free(ctx.spec);
 }
 
+TEST_F(QueryTest, testFieldSpec_v2) {
+  static const char *args[] = {"SCHEMA", "title",  "text", "weight", "0.1",    "body",
+                               "text",   "weight", "2.0",  "bar",    "numeric"};
+  QueryError err = {QUERY_OK};
+  IndexSpec *spec = IndexSpec_Parse("idx", args, sizeof(args) / sizeof(const char *), &err);
+  RedisSearchCtx ctx = SEARCH_CTX_STATIC(NULL, spec);
+  const char *qt = "@title:hello world";
+  QASTCXX ast(ctx);
+  int ver = 2;
+
+  ASSERT_TRUE(ast.parse(qt, ver)) << ast.getError();
+  //ast.print();
+  QueryNode *n = ast.root;
+  ASSERT_EQ(n->type, QN_PHRASE);
+  ASSERT_EQ(n->opts.fieldMask, -1);
+  ASSERT_EQ(n->children[0]->opts.fieldMask, 0x01);
```

Comment:
can you verify the number of children here? and types?

Reply:
improved. type of text leaves is always `QN_TOKEN` with or without modifier